auto-farm/src/external_sc_interactions/farm_actions.rs

```diff
@@ -60,6 +67,64 @@ pub trait FarmActionsModule:
         user_tokens_mapper.set(&new_user_farm_tokens);
     }
 
+    /// user_farm_ids contains the associated farm_id for each token in user_farm_tokens
+    fn compound_staking_rewards_with_existing_farm_position(
+        &self,
+        user: &ManagedAddress,
+        user_farm_tokens: &mut PaymentsVec<Self::Api>,
+        user_farm_ids: &ManagedVec<AddressId>,
+        new_tokens: EsdtTokenPayment,
+    ) -> Result<(), ()> {
+        let farm_id = self
+            .farm_for_farming_token(&new_tokens.token_identifier)
+            .get();
+        if farm_id == NULL_ID {
+            return Result::Err(());
+        }
+
+        let opt_existing_farm_index = user_farm_ids.find(&farm_id);
+        if opt_existing_farm_index.is_none() {
+            return Result::Err(());
+        }
+
+        let opt_farm_addr = self.farm_ids().get_address(farm_id);
+        if opt_farm_addr.is_none() {
+            return Result::Err(());
+        }
+
+        let existing_farm_index = unsafe { opt_existing_farm_index.unwrap_unchecked() };
+        let farm_addr = unsafe { opt_farm_addr.unwrap_unchecked() };
+        let existing_farm_pos = user_farm_tokens.get(existing_farm_index);
```

Comment:
Is there a chance for the `existing_farm_pos` variable to be at a different index than `existing_farm_index`, as they come from different storages?

Reply:
Nope, `user_farm_ids` is built using this function: https://github.com/multiversx/mx-auto-farm-sc/blob/dd4c4353630ac62d36ef8154e71bc3d015cf7183/auto-farm/src/whitelists/farms_whitelist.rs#L68

It's basically an optimization to not keep doing `self.farm_for_farm_token(user_pos.get(i).token_id).get()` every single time to get the farm's ID, and instead get it from the vec.